src/crates/mod.rs

```diff
@@ -74,50 +74,134 @@ impl TryFrom<&'_ PackageId> for Crate {
+                [_, _, "git", repo] => {
+                    if repo.starts_with("https://github.com") {
+                        Ok(Crate::GitHub(repo.replace('#', "/").parse()?))
+                    } else {
+                        let mut parts = repo.split('#').rev().collect::<Vec<_>>();
+                        let url = parts.pop();
+                        let sha = parts.pop();
+
+                        match (url, sha) {
+                            (Some(url), None) => Ok(Crate::Git(GitRepo {
+                                url: url.to_string(),
+                                sha: None,
+                            })),
+                            (Some(url), Some(sha)) => Ok(Crate::Git(GitRepo {
+                                // remove additional queries if the sha is present
+                                // as the crate version is already uniquely determined
+                                url: url.split('?').next().unwrap().to_string(),
+                                sha: Some(sha.to_string()),
+                            })),
+                            _ => bail!("malformed git repo: {}", repo),
+                        }
+                    }
+                }
+                _ => bail!(
+                    "malformed pkgid format: {}\n maybe the representation has changed?",
+                    pkgid.repr
+                ),
+            }
+        } else {
+            use cargo_util_schemas::core::*;
+
+            let package_id = PackageIdSpec::parse(&pkgid.repr)?;
 
-                    match (url, sha) {
-                        (Some(url), None) => Ok(Crate::Git(GitRepo {
-                            url: url.to_string(),
-                            sha: None,
-                        })),
-                        (Some(url), Some(sha)) => Ok(Crate::Git(GitRepo {
-                            // remove additional queries if the sha is present
-                            // as the crate version is already uniquely determined
-                            url: url.split('?').next().unwrap().to_string(),
-                            sha: Some(sha.to_string()),
-                        })),
-                        _ => bail!("malformed git repo: {}", repo),
+            match package_id.kind() {
+                Some(SourceKind::LocalRegistry) => Ok(Crate::Local(package_id.name().to_string())),
+                Some(SourceKind::Git(rev)) => {
+                    if let Some(url) = package_id.url() {
+                        if url.domain() == Some("github.com") {
+                            Ok(Crate::GitHub(GitHubRepo {
+                                org: url
+                                    .path_segments()
+                                    .and_then(|mut path| path.next())
+                                    .unwrap_or_default()
+                                    .to_string(),
+                                name: package_id.name().to_string(),
```

Comment:
Shouldn't this also be calculated from the URL? The name of the Cargo package might not match the repository name.

Reply:
Oh, `name` is supposed to be the repo name not the crate name, that makes sense.